community/ndp/messaging-v1/src/main/java/org/neo4j/ndp/messaging/v1/PackStreamMessageFormatV1.java

```diff
@@ -59,6 +61,45 @@
     public static final byte RELATIONSHIP = 'R';
     public static final byte PATH = 'P';
 
+    // The forward and backward step structures represent
+    // directed relationships within a path but do not
+    // contain identities for the start and end nodes of
+    // those relationships. This removes duplication of
+    // information within path structures, thereby reducing
+    // the overall size of a path's serialised representation.
+    //
+    // Example path:
+    //
+    //     (alice)-[:KNOWS]->(bob)<-[:KNOWS]-(carol)
+    //
+    // Structured representation:
+    //
+    // Struct 'P' Path {
+    //     Struct 'N' Node {           // (alice)
+    //         Identity, Labels, Properties
+    //     }
+    //     Struct '>' ForwardStep {    // -[:KNOWS]->
```

Comment:
One more thing also - you mentioned denormalizing paths, to avoid repeating the same node/rel data being sent over and over. Although perhaps that can be left as a future excercise.